src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7450,6 +7450,16 @@ or SyntaxKind.PlusPlusToken
                 or SyntaxKind.MinusMinusToken
                 or SyntaxKind.MinusGreaterThanToken;
 
+            // Now look for another set of items that indicate that we're not an attribute, but instead are a collection
+            // expression misplaced in an invalid top level expression-statement. (like `[] + b`).  These are
+            // technically invalid. But checking for this allows us to parse effectively to then give a good semantic
+            // error later on. These cases came from: ParseExpressionContinued
+            isCollectionExpression = isCollectionExpression
+                || IsExpectedBinaryOperator(this.CurrentToken.Kind)
+                || IsExpectedAssignmentOperator(this.CurrentToken.Kind)
+                || this.CurrentToken.Kind is SyntaxKind.DotDotToken
+                || this.CurrentToken.ContextualKind is SyntaxKind.SwitchKeyword or SyntaxKind.WithKeyword && this.PeekToken(1).Kind is SyntaxKind.OpenBraceToken;
```

Comment:
`as` and `is`?

Reply:
those are both covered by IsExpectedBinaryOperator.  I'll add explicit tests though.

Reply:
consider adding parens around the && expression.

Reply:
will do!